### PR TITLE
[plat-182] Inbound connections

### PR DESCRIPTION
**Pull request by raoulstrackx**
Enables the enclave to listen for incoming connections
This PR works together with [this rust PR](https://github.com/fortanix/rust/pull/3)

### CODE REVIEW

File: fortanix-vme/fortanix-vme-runner/src/server.rs [hunk 1]
```diff
@@ -0,0 +1,360 @@
+use fnv::FnvHashMap;
```
Comment (Pagten) [inline fortanix-vme/fortanix-vme-runner/src/server.rs:1]: Is this file an accidentally committed old version of what is now in `lib.rs`?

**Review by Pagten — APPROVED** [review]

File: fortanix-vme/fortanix-vme-runner/src/lib.rs [hunk 2]
```diff
@@ -186,61 +204,148 @@ impl Server {
...
+    }
+
+    fn add_listener_info(&self, info: ListenerInfo) -> RawFd {
+        let fd = info.listener.as_raw_fd();
+        self.listeners.lock().unwrap().insert(fd, Arc::new(Mutex::new(info)));
+        fd
+    }
+
+    fn listener_info(&self, fd: &RawFd) -> Option<Arc<Mutex<ListenerInfo>>> {
+        self.listeners.lock().unwrap().get(&fd).map(|m| m.clone())
```
[the remark this answers is not in the document]
Comment (raoulstrackx) [reply fortanix-vme/fortanix-vme-runner/src/lib.rs:218]: Oh good to know!

File: fortanix-vme/fortanix-vme-runner/src/lib.rs [hunk 3]
```diff
@@ -186,61 +204,148 @@ impl Server {
...
+            enclave.peer_port().unwrap_or_default(),
+            &format!("{:?}", &response),
+            Direction::Right,
+            "vsock");
+        enclave.write(&serde_cbor::ser::to_vec(&response).unwrap())?;
+        Ok(())
+    }
+
+    fn handle_request_accept(&self, fd: RawFd, enclave: &mut VsockStream) -> Result<(), IoError> {
+        let listener_info = self.listener_info(&fd).unwrap();
```
[the remark this answers is not in the document]
Comment (raoulstrackx) [reply fortanix-vme/fortanix-vme-runner/src/lib.rs:263]: Yes absolutely! That was an oversight

[the remark this answers is not in the document]
Comment (raoulstrackx) [reply fortanix-vme/fortanix-vme-runner/src/lib.rs:263]: Good catch!

File: fortanix-vme/fortanix-vme-runner/src/server.rs (hunk 1 above)
Comment (raoulstrackx) [reply fortanix-vme/fortanix-vme-runner/src/server.rs:1]: Yes it must have been introduced during a rebase. I updated the commit that introduced it

**Pull request merged**